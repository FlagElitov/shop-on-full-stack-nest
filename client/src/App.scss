body {
  margin: 0;
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen",
    "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue",
    sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

code {
  font-family: source-code-pro, Menlo, Monaco, Consolas, "Courier New",
    monospace;
}

.container {
  max-width: 1170px;
  margin: 0 auto;
  .cards {
    display: flex;
    margin-top: 80px;
    justify-content: space-around;
    .card {
      text-align: center;
      width: 400px;
      height: 600px;
      border: 3px #fafafa solid;
      box-shadow: 0px 0px 6px 0px;
      .name {
        margin-top: 100px;
        font-size: 51px;
      }
      .text {
        font-size: 28px;
        margin-top: 30px;
      }
      .balance {
        margin-top: 10px;
        font-size: 25px;
      }
      .input {
        margin-top: 30px;
      }
      .warning {
        font-size: 20px;
        color: red;
      }
      .button {
        margin-top: 60px;
      }
    }
  }
}
