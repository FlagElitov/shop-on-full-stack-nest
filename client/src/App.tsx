import axios from "axios";
import React from "react";
import "./App.scss";
import Card from "./component/Card/Card";

type dataTypes = {
  id: number;
  name: string;
  balance: number;
};

const App: React.FC = () => {
  const [data, setData] = React.useState<dataTypes[]>([
    { id: 1, name: "Artur", balance: 1000 },
    { id: 2, name: "Denis", balance: 1000 },
  ]);

  React.useEffect(() => {
    axios.get(`http://localhost:3005/products`).then((data) => {
      console.log(data);
    });
  }, []);

  const toggleHandler = (id: number) => {
    setData((prev) =>
      prev.map((data) => {
        if (data.id !== id) {
          return {
            ...data,
            balance: data.balance + 100,
          };
        }
        if (data.id === id) {
          return {
            ...data,
            balance: data.balance - 100,
          };
        }
        return data;
      })
    );
  };

  return (
    <div className="container">
      <div className="cards">
        {data.map((data: dataTypes) => (
          <Card
            key={data.id}
            id={data.id}
            toggleHandler={toggleHandler}
            name={data.name}
            balance={data.balance}
          />
        ))}
      </div>
    </div>
  );
};
export default App;
