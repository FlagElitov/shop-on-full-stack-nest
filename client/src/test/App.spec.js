import { shallow } from "enzyme";
import React from "react";
import App from "../App";

const setUp = () => shallow(<App />);

describe("Unit Testing App component", () => {
  let component;
  beforeEach(() => {
    component = setUp();
  });
  it("App Snapshot", () => {
    expect(component).toMatchSnapshot();
  });
});
