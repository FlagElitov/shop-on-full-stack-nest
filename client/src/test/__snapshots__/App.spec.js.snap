// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit Testing App component App Snapshot 1`] = `
<div
  className="container"
>
  <div
    className="cards"
  >
    <Card
      balance={1000}
      id={1}
      key="1"
      name="Artur"
      toggleHandler={[Function]}
    />
    <Card
      balance={1000}
      id={2}
      key="2"
      name="Denis"
      toggleHandler={[Function]}
    />
  </div>
</div>
`;
